fabric-levelgenerators-v1/src/main/java/net/fabricmc/fabric/mixin/levelgenerator/MixinServerPropertiesHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.levelgenerator;
+
+import java.util.Properties;
+import java.util.function.Function;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.server.dedicated.AbstractPropertiesHandler;
+import net.minecraft.server.dedicated.ServerPropertiesHandler;
+import net.minecraft.util.Identifier;
+import net.minecraft.world.level.LevelGeneratorType;
+
+import net.fabricmc.fabric.impl.levelgenerator.ServerPropertiesHandlerImplements;
+
+@Mixin(ServerPropertiesHandler.class)
+@Implements(@Interface(iface = ServerPropertiesHandlerImplements.class, prefix = "fabric$"))
```

Comment:
I'm not sure if this is using soft implements properly. Wouldn't that require having a `fabric$getFabricLevelType` method?